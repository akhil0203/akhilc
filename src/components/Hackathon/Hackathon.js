import React from 'react';

import './Hackathon.css';

const Hackathon = () => {
  return (
    <div className="Hackathon">
      <h1>HACKATHON</h1>
    </div>
  );
};

export default Hackathon;
