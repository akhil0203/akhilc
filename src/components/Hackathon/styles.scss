.hackathon-parallax {
  height: auto !important;
  background-size: cover;
}

.hackathon {
  background-size: cover;
  display: flex;

  > .col {
    padding: 150px 0 150px 0;
  }

  .col > .row:not(:last-child) {
    margin-bottom: 40px;
  }

  .row {
    justify-content: center;
    align-items: center;

    h2 {
      color: #fff;
      font-size: 40px;
      font-weight: 700;
      text-transform: uppercase;
    }

    h4 {
      color: #fff;
      font-size: 30px;
      font-weight: 700;
      text-align: center;
    }

    p {
      color: #fff;
      font-size: 16px;
      font-weight: 700;
      margin-bottom: 0;
      text-align: center;
    }

    img {
      width: 100%;
    }

    .winna {
      background-color: rgb(249, 242, 244);
      border-radius: 4px;
      color: rgb(199, 37, 78);
      padding: 2px 4px;
    }
  }
}
