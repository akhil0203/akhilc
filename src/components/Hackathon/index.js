import React from 'react';
import { Container, Col, Row } from 'reactstrap';
import { ParallaxBanner } from 'react-scroll-parallax';

import backgroundImg from '../../images/backgrounds/hackathon.jpg';

import tohacksImg from '../../images/hackathons/tohacks.png';
import hackingGoodImg from '../../images/hackathons/hacking-good.png';
import genesysImg from '../../images/hackathons/genesys.png';
import rbcNGIImg from '../../images/hackathons/rbc-ngi.png';
import openIDEOImg from '../../images/hackathons/open-ideo.png';
import gmHackAIImg from '../../images/hackathons/gm-hack-ai.png';
import osceDaysImg from '../../images/hackathons/osce-days.png';

import './styles.scss';

const Hackathon = () => {
  return (
    <ParallaxBanner
      className="hackathon-parallax"
      layers={[
        {
          image: JSON.stringify(backgroundImg),
          amount: 0.2,
        },
      ]}
    >
      <Container className="hackathon" fluid>
        <Col>
          <Row>
            <h2>Hackathons</h2>
          </Row>
          <Col>
            <Col>
              <Row>
                <Col sm={3}>
                  <h4>
                    <span className="winna">4th Place</span> - Genesys Hackathon
                  </h4>
                  <img src={genesysImg} alt="genesys" />
                </Col>
              </Row>

              <Row>
                <Col sm={3}>
                  <h4>
                    <span className="winna">2nd Place</span> - HackingGood 2017
                  </h4>
                  <img src={hackingGoodImg} alt="hackingGood" />
                </Col>
                <Col sm={3}>
                  <h4>Mentor & Judge - TOHacks 2018</h4>
                  <img src={tohacksImg} alt="toHacks" />
                </Col>
                <Col sm={3}>
                  <h4>RBC Next Great Innovator</h4>
                  <img src={rbcNGIImg} alt="rbcNGI" />
                </Col>
              </Row>

              <Row>
                <Col sm={3}>
                  <h4>OpenIDEO 2017</h4>
                  <img src={openIDEOImg} alt="openIDEO" />
                </Col>
                <Col sm={3}>
                  <h4>GM HackAI</h4>
                  <img src={gmHackAIImg} alt="gmHackAI" />
                </Col>
                <Col sm={3}>
                  <h4>
                    <span className="winna">3rd Place</span> - OSCE Days 2017
                  </h4>
                  <img src={osceDaysImg} alt="osceDays" />
                </Col>
              </Row>
            </Col>
          </Col>
        </Col>
      </Container>
    </ParallaxBanner>
  );
};

export default Hackathon;
