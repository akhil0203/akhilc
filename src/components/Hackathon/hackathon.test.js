import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Hackathon from '.';

it('matches the snapshot', () => {
  const tree = renderer.create(<Hackathon />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the Hackathon component without crashing', () => {
  shallow(<Hackathon />);
});

it('renders the main header', () => {
  const wrapper = shallow(<Hackathon />);

  expect(wrapper.find('h2').html()).toEqual('<h2>Hackathons</h2>');
});

it('renders an image for Genesys', () => {
  const wrapper = shallow(<Hackathon />);

  expect(
    wrapper
      .find('Col>Row>Col')
      .at(0)
      .html(),
  ).toEqual(
    '<div class="col-sm-2"><h4><span class="winna">4th Place</span> - Genesys Hackathon</h4><img src="genesys.png" alt="genesys"/></div>',
  );
});

it('renders an image for HackingGood', () => {
  const wrapper = shallow(<Hackathon />);

  expect(
    wrapper
      .find('Col>Row>Col')
      .at(1)
      .html(),
  ).toEqual(
    '<div class="col-sm-2"><h4><span class="winna">2nd Place</span> - HackingGood</h4><img src="hacking-good.png" alt="hackingGood"/></div>',
  );
});

it('renders an image for TOHacks', () => {
  const wrapper = shallow(<Hackathon />);

  expect(
    wrapper
      .find('Col>Row>Col')
      .at(2)
      .html(),
  ).toEqual(
    '<div class="col-sm-2"><h4>Mentor &amp; Judge - TOHacks</h4><img src="tohacks.png" alt="toHacks"/></div>',
  );
});

it('renders an image for RBC NGI', () => {
  const wrapper = shallow(<Hackathon />);

  expect(
    wrapper
      .find('Col>Row>Col')
      .at(3)
      .html(),
  ).toEqual(
    '<div class="col-sm-2"><h4>RBC Next Great Innovator</h4><img src="rbc-ngi.png" alt="rbcNGI"/></div>',
  );
});

it('renders an image for OpenIDEO', () => {
  const wrapper = shallow(<Hackathon />);

  expect(
    wrapper
      .find('Col>Row>Col')
      .at(4)
      .html(),
  ).toEqual(
    '<div class="col-sm-2"><h4>OpenIDEO</h4><img src="open-ideo.png" alt="openIDEO"/></div>',
  );
});

it('renders an image for GM HackAI', () => {
  const wrapper = shallow(<Hackathon />);

  expect(
    wrapper
      .find('Col>Row>Col')
      .at(5)
      .html(),
  ).toEqual(
    '<div class="col-sm-2"><h4>GM HackAI</h4><img src="gm-hack-ai.png" alt="gmHackAI"/></div>',
  );
});

it('renders an image for OSCE Days', () => {
  const wrapper = shallow(<Hackathon />);

  expect(
    wrapper
      .find('Col>Row>Col')
      .at(6)
      .html(),
  ).toEqual(
    '<div class="col-sm-2"><h4><span class="winna">3rd Place</span> - OSCE Days</h4><img src="osce-days.png" alt="osceDays"/></div>',
  );
});
