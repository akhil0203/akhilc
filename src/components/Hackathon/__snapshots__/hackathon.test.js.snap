// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className="hackathon container-fluid"
>
  <div
    className="col"
  >
    <div
      className="row"
    >
      <h2>
        Hackathons
      </h2>
    </div>
    <div
      className="col"
    >
      <div
        className="row"
      >
        <div
          className="col-sm-2"
        >
          <h4>
            <span
              className="winna"
            >
              4th Place
            </span>
             - Genesys Hackathon
          </h4>
          <img
            alt="genesys"
            src="genesys.png"
          />
        </div>
        <div
          className="col-sm-2"
        >
          <h4>
            <span
              className="winna"
            >
              2nd Place
            </span>
             - HackingGood 2017
          </h4>
          <img
            alt="hackingGood"
            src="hacking-good.png"
          />
        </div>
        <div
          className="col-sm-2"
        >
          <h4>
            Mentor & Judge - TOHacks 2018
          </h4>
          <img
            alt="toHacks"
            src="tohacks.png"
          />
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-sm-2"
        >
          <h4>
            RBC Next Great Innovator
          </h4>
          <img
            alt="rbcNGI"
            src="rbc-ngi.png"
          />
        </div>
        <div
          className="col-sm-2"
        >
          <h4>
            OpenIDEO 2017
          </h4>
          <img
            alt="openIDEO"
            src="open-ideo.png"
          />
        </div>
        <div
          className="col-sm-2"
        >
          <h4>
            GM HackAI
          </h4>
          <img
            alt="gmHackAI"
            src="gm-hack-ai.png"
          />
        </div>
        <div
          className="col-sm-2"
        >
          <h4>
            <span
              className="winna"
            >
              3rd Place
            </span>
             - OSCE Days 2017
          </h4>
          <img
            alt="osceDays"
            src="osce-days.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
